File: lovekiri_client/lib/models/travelog.dart
import 'package:json_annotation/json_annotation.dart';

part 'travelog.g.dart';

@JsonSerializable(fieldRename: FieldRename.snake, explicitToJson: true)
class Travelog {
  Travelog();

  @JsonKey(required: true, disallowNullValue: true)
  late String locationName;

  @JsonKey(required: true, disallowNullValue: true)
  late String address;

  @JsonKey(required: true, disallowNullValue: true)
  late double lat;

  @JsonKey(required: true, disallowNullValue: true)
  late double lng;
  
  @JsonKey(required: true, disallowNullValue: true)
  late String category;

  @JsonKey(required: true, disallowNullValue: true)
  late double dateTime;

  @JsonKey(required: true, disallowNullValue: true)
  late int rate;
  
  @JsonKey(required: true, disallowNullValue: true)
  late String description;

  String? imgUrl;

  factory Travelog.fromJson(Map<String, dynamic> json) => _$TravelogFromJson(json);
  Map<String, dynamic> toJson() => _$TravelogToJson(this);
}

File: lovekiri_server/model/Travelog.js
const mongoose = require("mongoose");

const travelogScheme = mongoose.Schema({
  location_name: {
    type: String,
    trim: false,
    maxLength: 120,
    required: true
  },
  address: {
    type: String,
    trim: false,
    required: true,
  },
  lat: {
    type: Number,
    required: true
  },
  lng: {
    type: Number,
    required: true
  },
  category: {
    type: String,
    required: true,
  },
  date_time: {
    type: Number,
    required: true,
  },
  rate: {
    type: Number,
    required: true,
  },
  description: {
    type: String,
    required: true,
  },
  img: {
    type: String
  }
});


const Travelog = mongoose.model('Travelog', travelogScheme);
module.exports = { Travelog };
